SpatialGDK/Source/SpatialGDK/Private/EngineClasses/RPCContainer.h

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Schema/RPCPayload.h"
+#include "SpatialConstants.h"
+
+#include "CoreMinimal.h"
+
+// TODO(Alex): remove this logic when singletons can be referenced without entity IDs (UNR-1456) - deprecated???
+struct FPendingRPCParams
+{
+	FPendingRPCParams(UObject* InTargetObject, UFunction* InFunction, SpatialGDK::RPCPayload&& InPayload, int InReliableRPCIndex = 0);
+	~FPendingRPCParams() = default;
+
+	TWeakObjectPtr<UObject> TargetObject;
+	UFunction* Function;
+
+	int ReliableRPCIndex;
+	SpatialGDK::RPCPayload Payload;
+};
+
+using FPendingRPCParamsPtr = TSharedPtr<FPendingRPCParams>;
+
+DECLARE_DELEGATE_RetVal_OneParam(bool, FProcessRPCDelegate, FPendingRPCParamsPtr)
+
+class FRPCContainer
+{
+public:
+	void QueueRPC(FPendingRPCParamsPtr Params, ESchemaComponentType Type);
+	void ProcessRPCs(const FProcessRPCDelegate& FunctionToApply);
+	bool ObjectHasRPCsQueuedOfType(const UObject* TargetObject, ESchemaComponentType Type);
+
+private:
+	using FQueueOfParams = TQueue<FPendingRPCParamsPtr>;
+	using FRPCMap = TMap<TWeakObjectPtr<const UObject>, TSharedPtr<FQueueOfParams>>;
```

Comment:
So this is a map from a weak object pointer to a shared pointer of a queue of shared pointers. Do we actually need all that and is ownership of all the shared pointers actually shared. This is quite a difficult thread to follow.

Reply:
Yes, I will change them to unique ptrs, since RPCContainer is expected to be the only owner.

Reply:
Changed FPendingRPCParamsPtr to be UniquePtr.